src/avro_json_encoder.erl

```diff
@@ -208,30 +233,30 @@ do_encode_type(#avro_fixed_type{} = T, EnclosingNamespace) ->
                   } = T,
   SchemaObjectFields =
     [ optional_field(namespace, ns(Namespace, EnclosingNamespace),
-                     ?NS_GLOBAL, fun encode_string/1)
+                     ?NS_GLOBAL, fun encode_string/1, Opt)
+    , {name, encode_name(Name, canon_ns(Namespace, EnclosingNamespace), Opt)}
     , {type, encode_string(?AVRO_FIXED)}
-    , {name, encode_string(Name)}
     , {size, encode_integer(Size)}
-    , optional_field(aliases,   Aliases,   [], fun encode_aliases/1)
+    , optional_field(aliases,   Aliases,   [], fun encode_aliases/1, Opt)
     | CustomProps
     ],
   lists:flatten(SchemaObjectFields).
 
 %% @private
-encode_field(Field, EnclosingNamespace) ->
+encode_field(Field, EnclosingNamespace, Opt) ->
   #avro_record_field{ name    = Name
                     , doc     = Doc
                     , type    = Type
                     , default = Default
                     , order   = Order
                     , aliases = Aliases} = Field,
-  [ {name, encode_string(Name)}
-  , {type, do_encode_type(Type, EnclosingNamespace)}
+  [ {name, encode_name(Name, EnclosingNamespace, Opt)}
```

Comment:
this is field name, are you sure field names should be full-names too ?

Reply:
Seems like it's not needed.